### PR TITLE
Providers: Fix init with broken link in plugin_cache_dir

### PR DESCRIPTION
**Pull request by Filirom1**
Hello

Fixes https://github.com/hashicorp/terraform/issues/27445.

Instead of raising and error, it display a warning (TF_LOG=trace) and continue analyzing the next providers.

Cheers
Romain

### CODE REVIEW

**Review by pselle — APPROVED** [review]
I'm approving this but I'd like to request a second opinion from another Core team member.

Having run into this today because I'd messed with my settings to replicate the reported issue, I have a better understanding now that the scanning failing init completely, and opaquely, when installing another provider, is difficult and unhelpful. (in this case where I was replicating, it was trying to install an AWS provider, but I'd messed with my plugin-cache dirs for the purpose of recreating this issue, and it was confusingly failing)

Another comment on the initial report https://github.com/hashicorp/terraform/issues/27445#issuecomment-761696962 has a helpful context of how this can happen in real-world cases.

Requesting second opinion because I'm not well-familiar with the plugin installer and cache processes.

**Review by mildwonkey — APPROVED** [review]
🎉 thank you, this has been driving me crazy 😂

**Pull request merged**
